lib/core/engine/iteration.js

```diff
@@ -61,6 +61,57 @@ class Iteration {
     this.asyncScriptsByCategory = asyncScriptsByCategory;
   }
 
+  async runPrePostScript(script, driver, context, commands) {
+    let requiresPrivilege = false;
+
+    if (typeof script === 'object') {
+      const scriptObject = script;
+      if (
+        typeof scriptObject.function !== 'function' ||
+        typeof scriptObject.requires !== 'object'
+      ) {
+        log.error(
+          'New-style pre/post script detected ' +
+            'but with invalid construction. ' +
+            'It will not execute.'
+        );
+        script = null;
+      } else {
+        log.info('Working with a new-style script.');
+        script = scriptObject.function;
+        requiresPrivilege =
+          scriptObject.requires.privilege === true ? true : false;
+      }
+    }
+    if (
+      requiresPrivilege &&
+      (typeof driver.getContext !== 'function' ||
```

Comment:
What is `typeof undefined`?  I.e., does this really work for Chrome, where this will be `undefined`?

Reply:
According to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof), `typeof undefined` is undefined. I confirmed this is implemented that way on both Chrome and Fx. Are we okay with that?

Reply:
Yep!